homeassistant/components/icloud/config_flow.py

```diff
@@ -64,6 +66,10 @@ def __init__(self):
                     vol.Required(
                         CONF_PASSWORD, default=user_input.get(CONF_PASSWORD, "")
                     ): str,
+                    vol.Required(
```

Comment:
Indeed I haven't did it in purpose but I think it's also fine.

Required string will check if not None and not empty.
Required boolean will check if not None

So the behavior will be the same since a boolean have two values and the field have a default.

As you want

Reply:
In the code base I see both Required and Optional with defaults